### DISCUSSION TITLE
Some dark magic going on there

### DISCUSSION THREAD

**Discussion by Wulfheart** [zond/godip#127] [General]
If I have a godip.Adjudicator I'd like to determine if it is a Build Order and which type is built. The only way I found was reflection. 😓 

````golang
if reflect.TypeOf(val) == reflect.TypeOf(orders2.BuildOrder) {
    vao := reflect.ValueOf(val)
    fv := vao.Elem().FieldByName("typ").String()
    fmt.Println(godip.UnitType(fv))
}
````

![grafik](https://user-images.githubusercontent.com/25671390/102220887-b75fba80-3ee1-11eb-92d2-d0407d387999.png)

**Comment by zond** [comment]
Adjudicator has Type() and Targets() that should help you out :)

**Reply by Wulfheart** [reply]
But this doesn’t return the UnitType. 
Sorry for not clarifying which Type I meant.

**Reply by zond** [reply]
Oh! You're right! I guess we never needed that - probably since we don't render based on the orders themselves, but based on the string slices used by the order parser.

Feel free to add an interface in the build.go file that provides the e.g. UnitType() func, then you could check it with interface testing instead of reflection. Would that be unhacky enough? :D
